src/KingSystem/World/worldShootingStarMgr.cpp

```diff
@@ -1 +1,158 @@
 #include "KingSystem/World/worldShootingStarMgr.h"
+#include <container/seadObjArray.h>
+#include <random/seadGlobalRandom.h>
+#include "KingSystem/ActorSystem/actActorCreator.h"
+#include "KingSystem/ActorSystem/actBaseProcHeapMgr.h"
+#include "KingSystem/ActorSystem/actInstParamPack.h"
+#include "KingSystem/GameData/gdtManager.h"
+#include "KingSystem/World/worldManager.h"
+
+namespace ksys::world {
+
+ShootingStarMgr::ShootingStarMgr() = default;
+
+ShootingStarMgr::~ShootingStarMgr() = default;
+
+void ShootingStarMgr::init_(sead::Heap* heap) {
+    if (Manager::instance()->isDemo())
+        sEnabled = false;
+    else
+        initSchedule();
+}
+
+// NON_MATCHING
+void ShootingStarMgr::initSchedule() {
+    u32 start = sShootingStarStartHour;
+    u32 end_idx = sShootingStarEndHour - 1;
+
+    if (sShootingStarStartHour >= sShootingStarEndHour) {
+        sead::ObjArray<u32> validHours;
```

Comment:
Does it match if you use a `sead::FixedObjArray<u32, 24>`? FixedObjArray takes care of setting a correctly sized buffer 

(Btw using a simple buffer of u32 is incorrect; the buffer size must be >= sead::ObjArray::calculateWorkBufferSize().)